src/checkers/inference/VariableAnnotator.java

```diff
@@ -1632,6 +1632,21 @@ private VariableSlot getOrCreateDeclBound(AnnotatedDeclaredType type) {
         return declSlot;
     }
 
+    /**
+     * Get the {@link VarAnnot} on the class declaration of the Element.
+     * @param ele an element
+     * @return the {@link VarAnnot} on the class declaration,
+     * or {@code null} if the class declaration of the Element is not handled by the
+     * {@link VariableAnnotator#getOrCreateDeclBound(AnnotatedDeclaredType)}.
+     */
+    public AnnotationMirror getClassDeclVarAnnot(Element ele) {
```

Comment:
1. parameter change to `TypeElement`
2. Consider the case that cache doesn't contain the type, return `VarAnnot` for top annotation

Reply:
For 2: consider use `AnnotatedTypeFactory#getDefaultTypeDeclarationBounds` instead of just using top.